ui-app/client/src/components/FormStorageEditor/components/formPreviewGeneration.ts

```diff
@@ -0,0 +1,115 @@
+import {
+	ComponentDefinition,
+	ComponentProperty,
+	DataLocation,
+	PageDefinition,
+} from '../../../types/common';
+import {
+	FormCompDefinition,
+	FormCompValidation,
+	FormPreviewCompValidation,
+	FormDefinition,
+	previewCompDefinitionMap,
+	previewCompValidationMap,
+	PreviewCompValidationProperties,
+} from './formCommons';
+import { duplicate } from '@fincity/kirun-js';
+
+const COMP_TYPE: { [key: string]: string } = {
+	NameEditor: 'TextBox',
+	EmailEditor: 'TextBox',
+	PhoneEditor: 'TextBox',
+	TextBoxEditor: 'TextBox',
+	TextAreaEditor: 'TextArea',
+	DropdownEditor: 'Dropdown',
+	RadioButtonEditor: 'RadioButton',
+	CheckBoxEditor: 'RadioButton',
+};
+
+const compValidationGenerator = (val: FormCompValidation) => {
+	let tempVal: FormPreviewCompValidation = {
+		...(duplicate(previewCompValidationMap.get(val.type)!) as FormPreviewCompValidation),
+		key: val.uuid,
+		order: val.order,
+	};
+	let valProps: PreviewCompValidationProperties = {
+		...tempVal.property.value,
+	};
+
+	if (val?.type) valProps['type'] = val?.type;
```

Comment:
Why do you use optional chaining here? if val is null or undefined this method should not and will not be called right?